internal/schema/connection.go

```diff
@@ -43,9 +45,22 @@ func GetEsFWConnectionBlock(keyName string) fwschema.Block {
 					Optional:            true,
 					Sensitive:           true,
 					Validators: []validator.String{
-						stringvalidator.ConflictsWith(usernamePath, passwordPath),
+						stringvalidator.ConflictsWith(usernamePath, passwordPath, bearerTokenPath),
 					},
 				},
+				"bearer_token": fwschema.StringAttribute{
+					MarkdownDescription: "Bearer Token to use for authentication to Elasticsearch",
+					Optional:            true,
+					Sensitive:           true,
+					Validators: []validator.String{
+						stringvalidator.ConflictsWith(usernamePath, passwordPath, apiKeyPath),
+					},
+				},
+				"es_client_authentication": fwschema.StringAttribute{
+					MarkdownDescription: "ES Client Authentication field to be used with the bearer token",
+					Optional:            true,
+					Sensitive:           true,
```

Comment:
```suggestion
					Sensitive:           true,
					Validators: []validator.String{
						stringvalidator.ConflictsWith(usernamePath, passwordPath, apiKeyPath),
						stringvalidator.AlsoRequires(bearerTokenPath),
					},
```

It only really makes sense for this to be used in conjunction with a bearer token IIUC.